### ISSUE TITLE
V0.5.0 Release Plan 

### ISSUE THREAD

**Issue by cp5555** [microsoft/superbenchmark#280]
# Release Manager
@cp5555 

# Endgame
Code freeze: April 10th, 2022
Bug Bash date: April 11th, 2022
Release date: April 22th, 2022

# Main Features
## Micro-benchmark Improvement 
1. - [x] Support nccl bandwidth benchmark only with NIC in NCCL/RCCL Bandwidth Test (#299)
2. - [x] Support bi-directional bandwidth benchmark in GPU Copy Bandwidth Test (#285, #298, #302)
3. - [x] Support data checking in GPU Copy Bandwidth Test (#301)
4. - [x] Update rccl-tests submodule to fix divide by zero error (#306)
5. - [x] Add GPU-Burn as microbenchmark (#324)

## Model-benchmark Improvement 
1. - [x] Sync results on root rank for e2e model benchmarks in distributed mode (#287) 
2. - [x] Support customized `env` in local and torch.distributed mode  (#295)
3. - [x] Add support for pytorch>=1.9.0 (#305)
4. - [x] Keep BatchNorm as fp32 for pytorch cnn models cast to fp16 (#322)
5. - [x] Remove FP16 samples type converting time (#330, #332)
6. - [x] Support FAMBench (#338)

## Inference Benchmark Improvement 
1. - [x] Revise the default setting for inference benchmark (#311, #329)
2. - [x] Add percentile metrics for inference benchmarks (#283)
3. - [x] Support T4 and A10 in GEMM benchmark (#294)
4. - [x] Add configuration with inference benchmark (#311)

## SuperBench Improvement 
1. - [x] Add command to support listing all optional parameters for benchmarks. (#279)
2. - [x] Unify benchmark naming convention and support multiple tests with same benchmark and different parameters/options in one configuration file (#284)
3. - [x] Support timeout to detect the benchmark failure and stop the process automatically (#288)
4. - [x] Add rocm5.0 dockerfile (#307)
5. - [x] Improve Output Interface (#333)

## Data Diagnosis & Analysis
1. - [x] Support multi-benchmark check (#289)
2. - [x] Support result summary in md, excel and html format （#320, #321, #335) 
3. - [x] Support data diagnosis in md and html format （#325) 
4. - [x] Support result output for all nodes in data diagnosis (#336, #339)
5. - [x] Add document for result summary usage (#337)

------------
# Backlogs
## SuperBench Improvement
1. - [ ] Support automatic configuration yaml selection on Azure VM

## Inference Benchmark Improvement
1. - [ ] Support VGG, LSTM, and GPT-2 small in TensorRT Inference Backend
2. - [ ] Support VGG, LSTM, and GPT-2 small in ORT Inference Backend

## Data Diagnosis & Analysis
1. - [ ] Support boxplot and outlier analysis

## Document
1. - [ ] Metric Reasoning Doc
